test/classes/initializers/postInit/args.good

```diff
@@ -0,0 +1 @@
+args.chpl:3: error: postInit must have zero arguments
```

Comment:
I'm not sure we should disallow `postInit()` methods with arguments since the user could choose to use that name for some other purpose (though stylistically, we'd discourage that).  We could create a style-/novice-oriented warning to point out the possibility of a mistake, like "note that the compiler won't insert automatic calls to this postInit() routine because of its required arguments."

We should also distinguish between those that have zero arguments and those that can be called with zero arguments (i.e., if `x` had a default value above, I think it'd be a valid `postInit()` for the compiler to call).

Reply:
I feel like it would be easier for users and ourselves if methods named 'postInit' were strictly limited to being a compiler hook. If a user wants to do something else, they should name it differently. From that point of view, I don't think there's much value in having a postInit where all formals have default values. It seems like an edge case not worth supporting (for now).

Would you be open to releasing with this error and relaxing later if demand is sufficient?

Reply:
(if it's simplest, I'd also be very OK (certainly for now) to just not have any kind of processing of postInit() routines whose signature is "surprising" to us.  This kindof relates to the conversation Vass and I were having the other day about the degree to which the compiler should prevent various signatures of init() and deinit() on modules.  One school of thought would be "anything that doesn't match what we're planning to call should be illegal because the user will only confuse themselves.  Another, and the one I'm leaning towards in both these conversations is, since those aren't reserved names, the compiler shouldn't prevent their use in other cases (but warnings might be useful to help prevent obvious mistakes)).

Reply:
Oops, I hadn't seen your comment before posting my follow-up

I suppose that what you've done is consistent with what we have for `deinit()`, so I wouldn't stand in its way.  In both cases, I do think that an error might be heavy-handed (though I also agree that the chances that a user cares very much is slim).

Reply:
OK. I'm happy to discuss this further if you want.